src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -258,126 +268,78 @@ class PeerActor(
       context stop self
   }
 
+  //FIXME: As the peer chain check is part of the handshaker, it should be incorporated into the handshake
```

Comment:
Maybe this was a wrongly written comment but the idea was to justify why this function was brought back (it was removed in https://github.com/input-output-hk/etc-client/commit/4979c7524105391c84de20f1f9296424b535b800#diff-d88dc0e9076bb9b1c58875187ebcd4acL270). I left a comment on PR #185 about this.